integration-tests/tests/run-notebook.js

```diff
@@ -24,7 +24,10 @@ const testRunNotebookFn = _.flow(
   await click(page, clickable({ text: 'Create Notebook' }))
   await click(page, clickable({ textContains: notebookName }))
   await click(page, clickable({ text: 'Edit' }))
-  await select(page, 'ENVIRONMENT', 'Hail')
+  // wait for drawer to slide in before clicking, otherwise we'll miss the button
+  await delay(1000)
```

Comment:
I'll be happier if that works. I think that was my original thought but ultimately went with this. I'll try `waitForNoSpinners`. (This is a small enough task for me today, and I want to see it myself given the context I already have with debugging the issue.)

Reply:
@panentheos Unfortunately it didn't work. This is really tricky to describe but I'll give it a try.

As a test case, I removed the slide-in and all delays between clicking "Edit" and "Customize". The test passes. I was curious to know what it thought it clicked. I instrumented capturing a screenshot of the element before clicking it. (This is not a perfect experiment because taking the screenshot takes >0 time.) Here's what it looks like:

![button_no_slide_in_no_delay](https://user-images.githubusercontent.com/19228696/92762732-e1dd2d00-f360-11ea-973c-53f5717914f2.png)

My interpretation of this is that puppeteer locates the element, calculates the rectangle, then doesn't capture the screenshot until after something happens that causes the button's location on the screen to shift. Since I've disabled the slide-in, the only remaining cause I can think of is something being returned from an ajax call affecting the rendering of the panel. In this case, the test still passes because (I think) puppeteer chooses to click the middle of the rectangle, which luckily still hits the button.

When I add `waitForNoSpinners`, I've seen the test fail with a button that looks like this:

![button_no_slide_wait_for_no_spinners](https://user-images.githubusercontent.com/19228696/92770123-9f6b1e80-f367-11ea-9588-f596649fdf2b.png)

This looks to me like the button is covered by a spinner overlay. My hunch here is that `waitForNoSpinners` is returning before the spinner appears. Then, by coincidence, the spinner was rendered at the time that the screenshot was taken and the click attempted.

For reference, here's something like what I saw when I originally started debugging this:

![button_slide_no_delay](https://user-images.githubusercontent.com/19228696/92772885-1c979300-f36a-11ea-9a6f-86924c5a8f28.png)

Finally, when I added `waitForNoSpinners`, the test failed when not capturing a screenshot of the button. When I captured a screenshot of the button, the test _passed_, but the button looked like this:

![button_slide_wait_for_not_spinners](https://user-images.githubusercontent.com/19228696/92777949-b2cdb800-f36e-11ea-8290-d163023f174f.png)

I did not run multiple experiments with each conditions to determine how stable these behaviors are, but I think the evidence is clear enough that the ideal code does not reliably produce the desired behavior.

My conclusion is that the cause is elements moving due to a combination of drawer sliding and rendering changes after ajax calls return. `await delay(1000)` is the only reliable fix because of a timing mismatch between spinner rendering and test execution.

Reply:
Thanks for the details investigation! I wonder if we should actually have *both*: first a time delay to wait for the animation, and then a spinner check to make sure the ajax calls have finished (commented for intent). That would be more resilient to network delays.

Reply:
Sure. It's better than just the delay. I have worries and I started to express them here, but it's getting too complicated and I just need to move forward.